packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,175 +70,307 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
+                    )
+                })?;
+                fns.set_index(scope, 1, undefined).ok_or_else(|| {
+                    Error::PackageImport(
+                        path.clone(),
+                        "Could not set Undefined value on package function array".to_string(),
+                    )
+                })?;
+                fns.set_index(scope, 2, undefined).ok_or_else(|| {
+                    Error::PackageImport(
+                        path.clone(),
+                        "Could not set Undefined value on package function array".to_string(),
+                    )
+                })?;
+
                 return Ok(JsPackage { fns });
             }
         };
 
         // Avoid JS ReferenceError by wrapping potentially undeclared variables with `typeof`.
-        // (Double braces like `{{` are Rust's escape string for a single literal `{`.)
-        let wrapped_code = format!(
-            "((hash_util, hash_stdlib)=>{{{}
-            return [
-                typeof start_experiment === 'undefined' ? undefined : start_experiment,
-                typeof start_sim === 'undefined' ? undefined : start_sim,
-                typeof run_task === 'undefined' ? undefined : run_task
-            ]}})",
-            code
-        );
-        let pkg: mv8::Function<'_> = mv8
-            .eval(wrapped_code)
-            .map_err(|e| Error::PackageImport(path.clone(), e.into()))?;
-        let args = mv8::Values::from_vec(vec![
-            embedded.hash_util.clone(),
-            embedded.hash_stdlib.clone(),
-        ]);
-
-        let fns: mv8::Array<'_> = pkg
-            .call(args)
-            .map_err(|e| Error::PackageImport(path.clone(), e.into()))?;
-        if fns.len() != 3 {
-            return Err(Error::PackageImport(path.clone(), "Stray return".into()));
+        // Double braces like `{{` are Rust's escape string for a single literal `{`.
+        let wrapped_code = new_js_string(
+            scope,
+            &format!(
+                "((hash_util, hash_stdlib) => {{{code}
+                    return [
+                        typeof start_experiment === 'undefined' ? undefined : start_experiment,
+                        typeof start_sim === 'undefined' ? undefined : start_sim,
+                        typeof run_task === 'undefined' ? undefined : run_task
+                    ]
+                }})",
+            ),
+        )?;
+
+        let pkg: v8::Local<'_, v8::Function> = {
+            let pkg = v8::Script::compile(scope, wrapped_code, None).ok_or_else(|| {
+                Error::PackageImport(
+                    path.clone(),
+                    format!("Could not compile code for package: {name}"),
+                )
+            })?;
+            let pkg = pkg.run(scope).ok_or_else(|| {
+                Error::PackageImport(
+                    path.clone(),
+                    format!("Couldn't execute package {name} setup"),
```

Comment:
Yeah apologies, this was my suggestion. I suggested we rework the messages to include the name because we were using `Error::V8` before. We don't need `{name}` now that it's `PackageImport` and we have `path`